### PR TITLE
switch tooling over to typespec

### PR DESCRIPTION
**Pull request by iscai-msft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by msyyc — CHANGES_REQUESTED** [review]
we shall wait for other language together and test it before merge so block it temporarily.
java: https://github.com/Azure/azure-sdk-for-java/pull/34136/files

File: tools/azure-sdk-tools/packaging_tools/sdk_generator.py
Comment (benbp) [inline tools/azure-sdk-tools/packaging_tools/sdk_generator.py:92]:
```suggestion
        spec_word = "typespecProject"
```

Comment (msyyc) [reply tools/azure-sdk-tools/packaging_tools/sdk_generator.py:92]: image

**Review by msyyc — APPROVED** [review]

File: eng/common/scripts/Cadl-Project-Sync.ps1
Comment (weshaggard) [inline eng/common/scripts/Cadl-Project-Sync.ps1:1]: You should not be deleting files under `eng\common` directly in a language repo. In fact I don't know how this didn't fail as we have a CI check for this. I've filed a bug to investigate this.

Comment (weshaggard) [reply eng/common/scripts/Cadl-Project-Sync.ps1:1]: I got this back in sync with https://github.com/Azure/azure-sdk-tools/issues/5882

**Pull request merged**
